tutorial/SparkRedshiftTutorial.scala

```diff
@@ -0,0 +1,172 @@
+			.load()
+		eventsDF.show()
+		eventsDF.printSchema()
+
+    /** Load from a query */
+		val salesQuery = """select salesid,listid,sellerid,buyerid,
+                        eventid,dateid,qtysold,pricepaid,commission 
+                        from sales 
+                        order by saletime desc LIMIT 10000"""
+		val salesDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL)
+			.option("tempdir", tempS3Dir)
+			.option("query", salesQuery)
+			.load()
+		salesDF.show()
+
+		val eventQuery = "select * from event"
+		val eventDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL) 
+			.option("tempdir", tempS3Dir)
+			.option("query", eventQuery)
+			.load()
+
+    /** Register 'event' table as 'myevent' in the Spark Environment */
+		eventsDF.registerTempTable("myevent") 
+
+    /** Save to a redshift table from a table registered in spark */
+    
+    /*
+     * Create a new table redshiftevent after dropping any existing redshiftevent table
+     * and write event records with event id less than 1000
+     */
+		sqlContext.sql("select * from myevent where eventid<=1000").withColumnRenamed("eventid", "id")
+			.write.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL)
+			.option("tempdir", tempS3Dir)
+			.option("dbtable", "redshiftevent")
+			.mode(SaveMode.Overwrite)
+			.save()
+    
+    /*
+     * Append to an existing table redshiftevent if it exists or create a new one if it does not
+     * exist and write event records with event id greater than 1000
+     */
+		sqlContext.sql("select * from myevent where eventid>1000").withColumnRenamed("eventid", "id")
+			.write.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL) 
+			.option("tempdir", tempS3Dir)
+			.option("dbtable", "redshiftevent")
+			.mode(SaveMode.Append) 
+			.save()
+
+		/** Demonstration of Inter-operability */
+		val salesAGGQuery = """select sales.eventid as id,sum(qtysold) as totalqty,sum(pricepaid)  as salesamt
+                           from sales
+                           group by (sales.eventid)
+                           """
+		val salesAGGDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL) 
+			.option("tempdir", tempS3Dir) 
+			.option("query", salesAGGQuery) 
+			.load()
+		salesAGGDF.registerTempTable("salesagg")
+
+    /*
+     * Join two DataFrame instances. Each could be sourced from any 
+     * compatible Data Source
+     */
+		val salesAGGDF2 = salesAGGDF.join(eventsDF, salesAGGDF("id") === eventsDF("eventid"))
+			.select("id", "eventname", "totalqty", "salesamt")
+
+    /** Create a custom schema */
+		val schema = StructType(List(StructField("eventid", IntegerType, false),
+			StructField("eventname", StringType, false),
+			StructField("qtysold", LongType, false),
+			StructField("revenue", DecimalType(38, 2), false)))
+      
+    /** Apply custom schema on existing DataFrame */
```

Comment:
Agreed. It does not add any value from the point of view of spark-redshift. Especially since Redshift does not care about the not-null directives and we can change the column name in the SELECT with an AS clause on columns. I re-wrote it as follows

While both `eventDF` and `salesAGGDF` table are sourced from Redshift, assume for the sake of illustration that they are sourced from a non-Redshift datasource. This should not be too far-fetched since the DataFrame class is part of the Data Sources API and not the `spark-redshift` library. A `DataFrame` can be created from any datasource compatible with the Data Sources API.

`salesAGGDF2` `DataFrame` is created by joining `eventsDF` and `salesAGGDF2`. We register it as a temporary table `redshift_sales_agg` before saving it to Redshift with the same name  `redshift_sales_agg`
  

```scala
salesAGGDF2.registerTempTable("redshift_sales_agg")

sqlContext.sql("SELECT * FROM redshift_sales_agg")
          .write.format("com.databricks.spark.redshift")
          .option("url", jdbcURL)
          .option("tempdir", tempS3Dir)
          .option("dbtable", "redshift_sales_agg")
          .mode(SaveMode.Overwrite)
          .save()
```